source/SIL.AppBuilder.Portal.Frontend/src/ui/components/inputs/locale-switch/index.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import { compose } from 'recompose';
+import { Dropdown } from 'semantic-ui-react';
+import { WithDataProps } from 'react-orbitjs';
+import { withTranslations, i18nProps } from '@lib/i18n';
+import { attributesFor } from '@data/helpers';
+import { UserAttributes } from '@data/models/user';
+import { ResourceObject } from 'jsonapi-typescript';
+
+import { USERS_TYPE, update } from '@data';
+
+import { withCurrentUserContext } from '@data/containers/with-current-user';
+
+export interface IOwnProps {
+  currentUser: ResourceObject<USERS_TYPE, UserAttributes>;
+  onChange: (locale: string) => void;
+  className?: string;
+}
+
+export type IProps = IOwnProps & WithDataProps & i18nProps;
+
+const translationMap = { 'en-US': 'en', 'es-419': 'es', 'fr-FR': 'fr' };
+
+class LocaleSwitch extends React.Component<IProps> {
+  onSelect = async (e, { value }) => {
+    e.preventDefault();
+
+    const { i18n } = this.props;
+    const { currentUser, dataStore } = this.props;
+
+    i18n.changeLanguage(value);
+
+    await update(dataStore, currentUser, {
+      attributes: { locale: value },
+    });
+  };
+
+  render() {
+    const { currentUser, i18n } = this.props;
+    const attributes = attributesFor(currentUser) as UserAttributes;
+    const userLocale = attributes.locale;
+    const { options, language } = i18n;
+    const languages = Object.keys(options.resources);
+
+    const selected = userLocale || language;
+
+    const languageOptions = languages.map((locale) => ({
+      text: translationMap[locale],
+      value: locale,
+    }));
+
+    return (
+      <Dropdown
+        data-test-locale-switcher
+        inline
+        options={languageOptions}
+        defaultValue={selected}
+        onChange={this.onSelect}
+      />
+    );
+  }
+}
+
+export default compose(
+  withTranslations,
+  withCurrentUserContext
+)(LocaleSwitch);
```

Comment:
for new components going forward (as of two weeks ago anyway), 
to get access to translations and current user, we should use "hooks":
like this:

```
export default function LocaleSwitch({ onChange, className } IProps) {
  const { currentUser } = useCurrentUser();
  const { i18n } = useTranslations();

  const onSelect = ...

  // the rest of the variables

  return (
    // same as in your render function, but no `this`
  );
}
```

this is because of a shift in the react programming paradigm, where classes are no longer used for components and higher-order components (withTranslations, etc) are used MUCH less frequently

Reply:
Rewrote this to match the comment